buildenv/jenkins/clang_format/Dockerfile

```diff
@@ -20,25 +20,45 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
 ###############################################################################
 
+# Some sections generated by Claude 3.7 Sonnet (version 20250219).
+
 FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
 
+# Create a non-root user and group.
+RUN microdnf install -y shadow-utils \
+    && groupadd -g 1000 clanguser \
+    && useradd -u 1000 -g clanguser -m -s /bin/bash clanguser \
+    && microdnf remove -y shadow-utils \
+    && microdnf clean all
+
+# Install required packages.
 RUN set -eux; \
     microdnf install -y \
        wget tar xz ncurses-compat-libs \
     && microdnf update -y \
-    && microdnf clean all; \
-    cd /tmp; \
-    wget https://github.com/llvm/llvm-project/releases/download/llvmorg-18.1.8/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04.tar.xz \
-    && tar -xvf clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04.tar.xz \
-    && mkdir -p /usr/bin \
-    && cp clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-format /usr/bin; \
-    rm -rf clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04*; \
-    microdnf remove wget tar xz
+    && microdnf clean all
 
-ENV PATH=/usr/bin:$PATH
+# Create directories with appropriate permissions.
+RUN mkdir -p /opt/clang-tools /src \
+    && chown -R clanguser:clanguser /opt/clang-tools /src
 
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+# Install clang-format in a location accessible to non-root users.
+RUN cd /tmp \
+    && wget https://github.com/llvm/llvm-project/releases/download/llvmorg-18.1.8/clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04.tar.xz \
+    && tar --no-same-owner -xvf clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04.tar.xz clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-format \
+    && cp clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04/bin/clang-format /opt/clang-tools/clang-format.real \
+    && rm -rf clang+llvm-18.1.8-x86_64-linux-gnu-ubuntu-18.04* \
+    && microdnf remove -y wget tar xz \
```

Comment:
Unless there's a reason to do otherwise (and I don't think there is), package names should be listed alphabetically.